client/post-editor/editor-image-selector/preview.jsx

```diff
@@ -0,0 +1,223 @@
+import { url } from 'lib/media/utils';
+
+export class EditorImageSelectorPreview extends React.Component {
+	static displayName = 'EditorImageSelectorPreview';
+
+	static propTypes = {
+		siteId: PropTypes.number,
+		itemIds: PropTypes.array,
+		multiple: PropTypes.bool,
+		maxWidth: PropTypes.number,
+		onImageChange: PropTypes.func,
+		showEditIcon: PropTypes.bool,
+		translate: PropTypes.func,
+	};
+
+	state = {
+		images: [],
+	};
+
+	componentDidMount() {
+		this.fetchImages();
+		MediaStore.on( 'change', this.updateImageState );
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { siteId, itemIds } = this.props;
+		if ( siteId !== prevProps.siteId || itemIds !== prevProps.itemIds ) {
+			this.fetchImages();
+		}
+	}
+
+	componentWillUnmount() {
+		MediaStore.off( 'change', this.updateImageState );
+	}
+
+	fetchImages = () => {
+		// We may not necessarily need to trigger a network request if we
+		// already have the data for the media item, so first update the state
+		this.updateImageState( () => {
+			if ( this.state.images ) {
+				return;
+			}
+
+			defer( () => {
+				this.props.itemIds.map( id => {
+					id = parseInt( id, 10 );
+					MediaActions.fetch( this.props.siteId, id );
+				} );
+			} );
+		} );
+	};
+
+	updateImageState = callback => {
+		if ( isEqual( this.state.images.map( image => image.ID ), this.props.itemIds ) ) {
+			return;
+		}
+
+		const images = uniq(
+			this.props.itemIds
+				.map( id => {
+					const media = MediaStore.get( this.props.siteId, id );
+					if ( ! media ) {
+						MediaActions.fetch( this.props.siteId, id );
+					}
+					return media;
+				} )
+				.filter( function( e ) {
+					return e;
+				} )
+		);
+
+		this.setState( { images }, () => {
+			if ( 'function' === typeof callback ) {
+				callback();
+			}
+		} );
+		defer( () => {
+			if ( this.props.onImageChange && images && images.length ) {
+				this.props.onImageChange( images );
+			}
+		} );
+	};
+
+	src = image => {
+		return url( image, {
+			maxWidth: this.props.maxWidth,
+			size: 'post-thumbnail',
+		} );
+	};
+
+	renderPlaceholder = image => {
+		const { placeholder } = image;
+		return (
+			<figure>
+				<img src={ placeholder || '' } alt="" />
+				<Spinner />
+			</figure>
+		);
+	};
+
+	renderUploaded = ( { URL, placeholder } ) => {
+		return (
+			<figure>
+				<ImagePreloader
+					src={ URL }
+					placeholder={ placeholder ? <img src={ placeholder } alt="" /> : <span /> }
+				/>
+			</figure>
+		);
+	};
+
+	renderImage = image => {
+		const src = this.src( image );
+		const { showEditIcon, translate } = this.props;
+		const id = image.ID || image.transientId;
+
+		const removeImage = () => {
+			this.props.onRemoveImage( image );
+		};
+
+		const classes = classNames( 'editor-image-selector__item', {
+			preview: null === src,
+		} );
+
+		return (
+			<div className={ classes } key={ id }>
+				<Button
+					className="editor-image-selector__image"
+					onClick={ this.props.onImageClick }
```

Comment:
could add `onImageClick` to line 135 to eliminate repeating `this.props` here.

Reply:
Destructured this and a few other props here.